src/tiled/propertybrowser.cpp

```diff
@@ -107,45 +107,13 @@ void PropertyBrowser::setObject(Object *object)
     if (mObject == object)
         return;
 
-    // Destroy all previous properties
-    mVariantManager->clear();
-    mGroupManager->clear();
-    mPropertyToId.clear();
-    mIdToProperty.clear();
-    mNameToProperty.clear();
-    mCustomPropertiesGroup = 0;
-
+    removeProperties();
     mObject = object;
 
     if (!mObject)
         return;
```

Comment:
Could you move this check also into `addProperties`? Then it can also be removed in `retranslateUi`.